edb/server/compiler_pool/server.py

```diff
@@ -0,0 +1,547 @@
+                    diff = client_schema.diff(cache)
+                if updated:
+                    # re-pickle the request if user schema changed
+                    msg = None
+                    extra_args = (method_name, args[0], *args[6:])
+            if msg:
+                msg = bytes(msg)
+            invalidation = worker.flush_invalidation()
+            resp = await worker.call(
+                "call_for_client",
+                client_id,
+                diff,
+                invalidation,
+                msg,
+                *extra_args,
+            )
+            status, *data = pickle.loads(resp)
+            if status == 0:
+                worker.set_client_schema(client_id, client_schema)
+                if method_name == "compile":
+                    units, new_pickled_state = data[0]
+                    if new_pickled_state:
+                        sid = worker._last_pickled_state = next_tx_state_id()
+                        resp = pickle.dumps((0, (*data[0], sid)), -1)
+            elif status == 1:
+                exc, tb = data
+                if not isinstance(exc, state_mod.FailedStateSync):
+                    worker.set_client_schema(client_id, client_schema)
+            else:
+                exc = RuntimeError(
+                    "could not serialize result in worker subprocess"
+                )
+                exc.__formatted_error__ = data[0]
+                raise exc
+
+            return resp
+        finally:
+            self._release_worker(worker)
+
+    async def compile_in_tx(
+        self, pickled_state, state_id, txid, *compile_args, msg=None
+    ):
+        if pickled_state == state_mod.REUSE_LAST_STATE_MARKER:
+            worker = await self._acquire_worker(
+                condition=lambda w: (w._last_pickled_state == state_id)
+            )
+            if worker._last_pickled_state != state_id:
+                print('cache miss')
+                self._release_worker(worker)
+                raise state_mod.StateNotFound()
+            else:
+                print('cache hit')
+        else:
+            print('no cache')
+            worker = await self._acquire_worker()
+        try:
+            resp = await worker.call(
+                "compile_in_tx", pickled_state, txid, *compile_args, msg=msg
+            )
+            status, *data = pickle.loads(resp)
+            if status == 0:
+                state_id = worker._last_pickled_state = next_tx_state_id()
+                resp = pickle.dumps((0, (*data[0], state_id)), -1)
+            return resp
+        finally:
+            self._release_worker(worker, put_in_front=False)
+
+    async def _request(self, method_name, msg):
+        worker = await self._acquire_worker()
+        try:
+            return await worker.call(method_name, msg=msg)
+        finally:
+            self._release_worker(worker)
+
+    async def call(self, protocol, req_id, msg):
```

Comment:
I'm confused... This isn't a "call an RPC", this is "handle a call", right? If so it should be named something like "handle_client_call", no?

Reply:
Yes!